website/static/js/projectSettings.js

```diff
@@ -7,84 +7,88 @@ var ko = require('knockout');
-            self.INSTANTIATION_ERROR_MESSAGE = 'Trying to instantiate NodeCategorySettings view model without an update URL';
+            self.title = ko.observable(params.currentTitle).extend({
+                required: {
+                    params: true,
+                    message: 'Title cannot be blank.'
+                }});
+            self.description = ko.observable(params.currentDescription);
+            self.titlePlaceholder = params.currentTitle;
+            self.descriptionPlaceholder = params.currentDescription;
 
-            self.MESSAGE_SUCCESS_CLASS = 'text-success';
-            self.MESSAGE_ERROR_CLASS = 'text-danger';
+            self.disabled = params.disabled || false;
 
-            if (!updateUrl) {
-                throw new Error(self.INSTANTIATION_ERROR_MESSAGE);
+            if (!params.updateUrl) {
+                throw new Error(language.instantiationErrorMessage);
             }
 
-            self.categories = categories;
-            self.category = ko.observable(category);
-            self.updateUrl = updateUrl;
+            self.categoryOptions = params.categoryOptions;
+            self.categoryPlaceholder = params.category;
+            self.selectedCategory = ko.observable(params.category);
 
-            self.selectedCategory = ko.observable(category);
-            self.dirty = ko.observable(false);
-            self.selectedCategory.subscribe(function(value) {
-                if (value !== self.category()) {
-                    self.dirty(true);
-                }
-            });
-        },
-        updateSuccess: function(newcategory) {
-            var self = this;
-            self.changeMessage(self.UPDATE_SUCCESS_MESSAGE, self.MESSAGE_SUCCESS_CLASS);
-            self.category(newcategory);
-            self.dirty(false);
+            self.updateUrl = params.updateUrl;
+            self.node_id = params.node_id;
         },
+        /*error handler*/
         updateError: function(xhr, status, error) {
             var self = this;
-            self.changeMessage(self.UPDATE_ERROR_MESSAGE, self.MESSAGE_ERROR_CLASS);
-            Raven.captureMessage(self.UPDATE_ERROR_MESSAGE_RAVEN, {
+            self.changeMessage(language.updateErrorMessage, 'text-danger');
+            Raven.captureMessage(language.updateErrorMessage, {
                 url: self.updateUrl,
                 textStatus: status,
                 err: error
             });
         },
-        updateCategory: function() {
+        /*update handler*/
+        updateAll: function() {
             var self = this;
-            return $osf.putJSON(self.updateUrl, {
-                    category: self.selectedCategory()
-                })
-                .then(function(response) {
-                    return response.updated_fields.category;
-                })
-                .done(self.updateSuccess.bind(self))
-                .fail(self.updateError.bind(self));
+            return $osf.ajaxJSON('put', self.updateUrl,
```

Comment:
Because you're attaching handlers etc, please break this into separate lines:

```
var request = $osf.ajaxJSON(...);
request.done(...);
request.fail(...);
return request;
```